### ISSUE TITLE
SIGCONT not triggering core dump until shutdown

### ISSUE THREAD

**Issue by dharmab** [fluent/fluent-bit#2109]
## Bug Report

**Describe the bug**

When I send a `CONT` signal to Fluent Bit, `[engine] caught signal (SIGCONT)` is printed to stdout immediately but the core dump is not dumped until after the engine stops.

**To Reproduce**

- Issued `kill -CONT $pid` to the Fluent Bit process:

```
[engine] caught signal (SIGCONT)
```

- Issued `kill -CONT $pid && sleep 15 && kill -TERM $pid`:

```
[engine] caught signal (SIGCONT)
[engine] caught signal (SIGTERM)
[2020/04/16 19:08:53] [ info] [input] pausing tail.0
[2020/04/16 19:08:53] [ info] [input] pausing storage_backlog.1
[2020/04/16 19:08:53] [ warn] [engine] service will stop in 5 seconds
[2020/04/16 19:08:57] [ info] [engine] service stopped
[2020/04/16 19:08:38] Fluent Bit Dump

===== Input =====
tail.0 (tail)
│
├─ status
│  └─ overlimit     : no
│     ├─ mem size   : 0b (0 bytes)
│     └─ mem limit  : 0b (0 bytes)
│
├─ tasks
│  ├─ total tasks   : 0
│  ├─ new           : 0
│  ├─ running       : 0
│  └─ size          : 0b (0 bytes)
│
└─ chunks
   └─ total chunks  : 0
      ├─ up chunks  : 0
      ├─ down chunks: 0
      └─ busy chunks: 0
         ├─ size    : 0b (0 bytes)
         └─ size err: 0

storage_backlog.1 (storage_backlog)
│
├─ status
│  └─ overlimit     : no
│     ├─ mem size   : 0b (0 bytes)
│     └─ mem limit  : 0b (0 bytes)
│
├─ tasks
│  ├─ total tasks   : 0
│  ├─ new           : 0
│  ├─ running       : 0
│  └─ size          : 0b (0 bytes)
│
└─ chunks
   └─ total chunks  : 0
      ├─ up chunks  : 0
      ├─ down chunks: 0
      └─ busy chunks: 0
         ├─ size    : 0b (0 bytes)
         └─ size err: 0


===== Storage Layer =====
total chunks     : 0
├─ mem chunks    : 0
└─ fs chunks     : 0
   ├─ up         : 0
   └─ down       : 0
```

Not shown: 15 second gap between signals

**Expected behavior**

The core dump should be printed to stdout immediately when `CONT` is received` instead of after the engine shuts down.

**Screenshots**

Note the iTerm timestamps on the right margin:

![Screen Shot 2020-04-16 at 13 10 35](https://user-images.githubusercontent.com/4450708/79496671-b0e01280-7fe3-11ea-9bad-affc94d935ab.png)

**Your Environment**

* Version used: 1.4.2
* Configuration: Complex configuration using input tail, file storage database buffer, many filters and output to Splunk. Redacted configuration snippets available on request
* Environment name and version (e.g. Kubernetes? What version?): Kubernetes 1.16.8
* Server type and version: 
* Operating System and version: Flatcar Linux (recent version)
* Filters and plugins:

**Additional context**

Found it while trialing upgrade from Fluent Bit 1.3 to Fluent Bit 1.4. This doesn't block our upgrade but the dumps would come in handy for debugging.

**Comment by edsiper** [comment]
the SIGCONT signal only prints the internals, no core dump will be generated.

**Comment by dharmab** [comment]
@edsiper thanks for taking a look. https://docs.fluentbit.io/manual/administration/dump-internals-signal#usage has a an example of the dump printing immediately after the `CONT` signal with no `TERM` required. Am I missing something?

**Comment by edsiper** [comment]
we handle signals in different ways, ```CONT``` is used to print (or dump a string) of informative stats of the input plugins, data ingestion, and the storage layer, this is useful for troubleshooting any time on runtime, no need to stop the service.

**Comment by dharmab** [comment]
Yes, that's my bug report- the `CONT` signal doesn't print anything until the service stops at a later point, when I expected it to print immediately.

**Comment by edsiper** [comment]
we send the data to stdout, are you sure Docker is not buffering stdout ?

**Comment by dharmab** [comment]
I'm fairly sure- when any of my other containers log a single short line to stdout, I see the output immediately in `docker logs` and `kubectl logs`.

**Comment by edsiper** [comment]
Fixed on #2176
